.github/workflows/test.yml

```diff
@@ -0,0 +1,24 @@
+name: Test
+on:
+  pull_request:
+  push:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [18.x]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn
+      - name: Build
+        run: yarn prepublish
+      - name: Run typecheck
+        run: yarn type-check
```

Comment:
It's unnecessary because command `yarn` already runs both `yarn prepublish` and `yarn type-check`